sql/2020/07_SEO/pages_wpt_bodies_text_vs_images_links_by_device.sql

```diff
@@ -0,0 +1,45 @@
+#standardSQL
+# text v image links grouped by device
+
+# helper to create percent fields
+CREATE TEMP FUNCTION AS_PERCENT (freq FLOAT64, total FLOAT64) RETURNS FLOAT64 AS (
+  ROUND(SAFE_DIVIDE(freq, total), 4)
+);
+
+# returns all the data we need from _wpt_bodies
+CREATE TEMPORARY FUNCTION get_wpt_bodies_info(wpt_bodies_string STRING)
+RETURNS STRUCT<
+    image_links INT64, 
+    text_links INT64 
+> LANGUAGE js AS '''
+var result = {};
+try {
+    var wpt_bodies = JSON.parse(wpt_bodies_string); 
+
+    if (Array.isArray(wpt_bodies) || typeof wpt_bodies != 'object') return result;
+
+    if (wpt_bodies.anchors && wpt_bodies.anchors.rendered) {
+
+      result.image_links = wpt_bodies.anchors.rendered.image_links;
+      result.text_links = wpt_bodies.anchors.rendered.text_links;
+    }
+
+} catch (e) {}
+return result;
+''';
+
+SELECT
+client,
+COUNT(0) AS total, 
+ROUND(AVG(wpt_bodies_info.image_links)) AS image_links_avg,
```

Comment:
Ditto on percentiles

Reply:
@antoineeripret I'm not sure how this one works?

Reply:
@Tiggerito : what I did there, as I was aware that a freq table would end up with a lot of possible values, was to apply an AVG() function on the returned results. Maybe not the best idea I have had. 

We can change it to return percentiles instead. I'll have a look at that during the afternoon (Spanish time).

Reply:
I've already added percentiles for those values in the percentiles script. Just want to make sure we can remove these?

Reply:
Ah, my bad! Yes, for sure !

Reply:
I've added some image v text link percentiles and dropped all the queries that are now covered by percentiles.

The requirements for "image v text links" are vague, in fact that is the requirements. I've created separate percentiles for each as well as a percentile of the percent of links that are images. Hopefully that will cover what they asked for.

I think all is done.